tests/unit/acquisition/test_optimizer.py

```diff
@@ -55,46 +60,91 @@ def _quadratic_sum(shift: list[float]) -> AcquisitionFunction:
             Box([-1], [2]),
             [1.0],
             [[1.0]],
-            [optimize_continuous, generate_random_search_optimizer(10_000)],
+            [generate_random_search_optimizer(10_000)],
         ),  # 1D
         (
             Box([-1, -2], [1.5, 2.5]),
             [0.3, -0.4],
             [[0.3, -0.4]],
-            [optimize_continuous, generate_random_search_optimizer(10_000)],
+            [generate_random_search_optimizer(10_000)],
         ),  # 2D
         (
             Box([-1, -2], [1.5, 2.5]),
             [1.0, 4],
             [[1.0, 2.5]],
-            [optimize_continuous, generate_random_search_optimizer(10_000)],
+            [generate_random_search_optimizer(10_000)],
         ),  # 2D with maximum outside search space
-        (
-            Box([-1, -2, 1], [1.5, 2.5, 1.5]),
-            [0.3, -0.4, 0.5],
-            [[0.3, -0.4, 1.0]],
-            [optimize_continuous, generate_random_search_optimizer(100_000)],
-        ),  # 3D
     ],
 )
-def test_optimizer(
+def test_discrete_and_random_optimizer(
     search_space: DiscreteSearchSpace,
     shift: list[float],
     expected_maximizer: list[list[float]],
     optimizers: list[AcquisitionOptimizer],
 ) -> None:
     for optimizer in optimizers:
         maximizer = optimizer(search_space, _quadratic_sum(shift))
-        if optimizer is optimize_continuous:
-            npt.assert_allclose(maximizer, expected_maximizer, rtol=1e-3)
-        elif optimizer is optimize_discrete:
+        if optimizer is optimize_discrete:
             npt.assert_allclose(maximizer, expected_maximizer, rtol=1e-4)
         else:
             npt.assert_allclose(maximizer, expected_maximizer, rtol=1e-1)
 
 
+def test_generate_continuous_optimizer_raises_with_invalid_init_params() -> None:
+    with pytest.raises(ValueError):
+        generate_continuous_optimizer(num_samples=-5)
+    with pytest.raises(ValueError):
+        generate_continuous_optimizer(num_restarts=-5)
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "search_space, shift, expected_maximizer",
+    [
+        (
+            Box([-1], [2]),
+            [1.0],
+            [[1.0]],
+        ),  # 1D
+        (
+            Box([-1, -2], [1.5, 2.5]),
+            [0.3, -0.4],
+            [[0.3, -0.4]],
+        ),  # 2D
+        (
+            Box([-1, -2], [1.5, 2.5]),
+            [1.0, 4],
+            [[1.0, 2.5]],
+        ),  # 2D with maximum outside search space
+        (
+            Box([-1, -2, 1], [1.5, 2.5, 1.5]),
+            [0.3, -0.4, 0.5],
+            [[0.3, -0.4, 1.0]],
+        ),  # 3D
+    ],
+)
+@pytest.mark.parametrize(
+    "optimizer",
+    [
+        generate_continuous_optimizer(),
```

Comment:
(Alternatively you could parameterize on `num_restarts` and `sigmoid` and generate the optimizer for the different combinations in the test.)